### ISSUE TITLE
html commenting by keyboard is adding //

### ISSUE THREAD

**Issue by Ahmed-E-Elswerkey** [Microsoft/vscode#70614]
- VSCode Version: latest
- OS Version: windows 10

Steps to Reproduce:

1.forget a semi-colon in a script tag in head
2.press ctrl + /

Does this issue occur when all extensions are disabled?: Yes

**Comment by roblourens** [comment]
If your cursor is inside the JS script tag then `//` is correct. Please explain your issue some more.

**Comment by Ahmed-E-Elswerkey** [comment]
the cursor is outside the script tag, html elements being commented by "//" because I forgot a ";" in the single line script tag

**Comment by roblourens** [comment]
Can you include a code sample or screenshot or something, I can't reproduce this

**Comment by Ahmed-E-Elswerkey** [comment]
![image](https://user-images.githubusercontent.com/18557802/54480349-fbff8300-482f-11e9-971b-11c5d84540ec.png)
![image](https://user-images.githubusercontent.com/18557802/54480351-0f125300-4830-11e9-872a-98b4a4d85d66.png)


sorry, I thought it could be easily reproduced 

so here, when missing a semicolon, the html is commented by // 
but when the semicolon is added, the html is commented the right way

**Comment by roblourens** [comment]
Ah I see it in that case. I think it will parse the opening `<` as a JS operator. But probably works in a browser.

**Comment by aeschli** [comment]
We use the grammar to detect embedded ranges, and the HTML grammar embeds the JavaScript grammar which isn't aware that it's embedded.

A fix for https://github.com/textmate/html.tmbundle/issues/85 would fix that

Alternative would be to not use text mate to detect embedded ranges.

https://github.com/Microsoft/vscode/issues/20488

**Comment by msftrncs** [comment]
I think that https://github.com/textmate/html.tmbundle/issues/85 will actually be fixed by https://github.com/Microsoft/vscode-textmate/pull/85, which I think was released this morning to 4.1.1 of the VS Code TextMate engine.   Hopefully we'll see this in insiders tonight and can check it out.

**Comment by msftrncs** [comment]
I don't think this issue will be fixed by https://github.com/Microsoft/vscode-textmate/pull/85, as I look at this, this specific issue is a limitation of TextMate, which lacks a construct that acts something between the Begin/End rules and the Begin/While rules.   Begin/While rules solve a very specific problem, but cannot be used here.  We really could use a Begin\Until system.  Unlike a Begin\While rule, a Begin\Until rule would capture all the content of the line up to the point the Until pattern matched and process it with the current rule stack, before restoring the stack to the condition it was in before the Begin/Until rule began.
